File: hashing/0219-contains-duplicate-ii/cpp/Solution.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> lastSeenIndex;

        for (int i = 0; i < (int)nums.size(); i++) {
            auto it = lastSeenIndex.find(nums[i]);
            if (it != lastSeenIndex.end() && i - it->second <= k) {
                return true;
            }
            lastSeenIndex[nums[i]] = i;
        }

        return false;
    }
};

File: hashing/0219-contains-duplicate-ii/java/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> lastSeenIndex = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            Integer prevIndex = lastSeenIndex.get(nums[i]);
            if (prevIndex != null && i - prevIndex <= k) {
                return true;
            }
            lastSeenIndex.put(nums[i], i);
        }

        return false;
    }
}
